### DISCUSSION TITLE
Geführtes Laden

### DISCUSSION THREAD

**Discussion by TheMikeCAD** [hoylabs/OpenDTU-OnBattery#1086] [Ideas]
Hallo zusammen,
hier mal als Idee / zur Diskussion:
Da ich verhindern möchte, dass die Anlage abregelt, wenn der Akku voll ist, habe ich ein "geführtes Laden" über IOBroker realisiert:
Es wird in Intervallen geprüft, wie voll der Akku ist und wie stark die Sonneneinstrahlung über die letzten X Minuten war. Abhängig davon wird das mit einer rudimentären "Ladekurve" (Bei mir sind es nur ein paar Punkte) verglichen.
Beispiel: Akku am Morgen 50% noch voll, hohe Einstrahlung -> Wechselrichter Vollgas (max. Netzeinspeisung), Akku "leer", passthrough hochsetzen bis Ladekurve; Akku voll, aber schwache Einstrahlung, passthrough hoch.. 
Braucht sowas außer mir noch jemand?
Wie gesagt, läuft das aktuell über IOBroker..

Gruß

**Comment by spcqike** [comment]
Ich habe über sowas auch schon nachgedacht. Aber hatte einen anderen Ansatz. 

Es gibt APIs die den zu erwartenden Tagesertrag ausspucken. Bspw 
https://api.forecast.solar/estimate/52.4/13.6/4/-51/3.4

Mein Gedanke war „bekommt man es hin, die batterieunterstützung so zu steuern, dass auch tagsüber möglichst viel aus der Batterie unterstützt wird, wenn nötig, aber trotzdem irgendwann die Batterie nur noch zu laden, so dass sie abends / nachts möglichst voll ist“? 

Die APi liefert Tages und Stunden Werte. Grobe Richtwerte, ohne Berücksichtigung des Wetters / Wolken. Bzw nur sehr knapp. 

Und ich selber habe noch gar keine Batterie (: von daher hab ich die Überlegungen erstmal wieder auf Eis gelegt.

**Comment by TheMikeCAD** [comment]
Damit hatte ich auch schon rumgespielt - ist mir aber nicht zielführend gelungen. Mein Ansatz es etwas rudimentärer - funktioniert aber in ca. 80% der Fälle.
Dein Ansatz wäre für meinen Anwendungsfall besser, da ich tagsüber keinen Strom von der Balkonanlage entnehme. Wenn Du aber hier wechselnde, unvorhersehbare Lasten hast, wird es schnell komplex fürchte ich. Bei meinem Ansatz wäre das ziemlich egal, da ich nur auf Batteriestand und Stromwerte (incl. Verschattung, Wolken etc.) reagiere.
